pystiche_papers/ulyanov_et_al_2016/_data.py

```diff
@@ -186,71 +185,74 @@ def images() -> DownloadableImageCollection:
     return DownloadableImageCollection({**content_images, **style_images})
 
 
+class SkipSmallIterableDataset(IterableDataset):
+    def __init__(
+        self,
+        dataset: Dataset,
+        min_size: int,
+        num_samples: int,
+        transform: transforms.Transform,
+    ):
+        self.dataset = dataset
+        self.min_size = min_size
+        self.num_samples = num_samples
+        self.transform = transform
+        self.data_samples = itertools.cycle(self.dataset)
+
+    def __len__(self) -> int:
+        return self.num_samples
+
+    def __iter__(self) -> Generator:
+        num_samples = 0
+        while num_samples < self.num_samples:
+            sample = next(self.data_samples)
+            if all(size >= self.min_size for size in extract_image_size(sample)):
+                # Images that are too small are skipped by the original DataLoader and
+                # the next image is used.
+                # https://github.com/pmeier/texture_nets/blob/aad2cc6f8a998fedc77b64bdcfe1e2884aa0fb3e/dataloader.lua#L91-L100
+                yield self.transform(sample)
+                num_samples += 1
+
+
 def dataset(
     root: str,
     impl_params: bool = True,
     instance_norm: bool = True,
     transform: Optional[transforms.Transform] = None,
-) -> ImageFolderDataset:
+    hyper_parameters: Optional[HyperParameters] = None,
+) -> SkipSmallIterableDataset:
     if transform is None:
```

Comment:
`content_transform()` also takes `hype_parameters`. Thus, we need to create it before and pass it on.